services/horizon/internal/render/problem/problem.go

```
@@ -8,6 +8,16 @@ import (
 
 // Well-known and reused problems below:
 var (
+	// Cancelled is a well-known problem type.  Use it as a shortcut
+	// in your actions.
+	Cancelled = problem.P{
+		Type:   "cancelled",
+		Title:  "Cancelled",
+		Status: http.StatusNoContent,
+		Detail: "The request has been cancelled due to user action or application " +
+			"shut down.",
```

I think we should change the message to more generically indicate that the service is unavailable. The message as it stands is a little confusing from a clients point-of-view.

For the first reason given, `due to user action`:
What actions can a client take that will cause a cancel? I don't think there are any. The server might cancel the query because the query was taking too long, or because the request as a whole was taking too long, but from the perspective of the Horizon client that's not the user's choice. Its boundaries we've set in Horizon to make Horizon unavailable to the client when those limits are reached.

For the second reason given, `due to...application shut down`:
An application shutting down is odd thing to communicate back to a client. We should be making our application shutdown gracefully such that connections are given enough time to complete inside the `WriteTimeout` (once we have that set). If a connection is dropped I think we can just say the service was unavailable in any case. We don't know who the clients for Horizon will be, and I don't think we should expose insight into what a Horizon host is doing.

Maybe:
>"The request cannot be serviced at this time."

---

> What actions can a client take that will cause a cancel? I don't think there are any.

It actually is, you can disconnect before response is sent to you. In such case, client would not see a response so I'm fine with removing this from error message.

> We should be making our application shutdown gracefully such that connections are given enough time to complete

Yes, we have graceful server in Horizon, the problem is that the timeout is currently set to 10 seconds. So if processing request takes longer than that we need the context will be cancelled and we'll need to return an error.

But I agree with both points, I'm OK with changing the message to: `The request cannot be serviced at this time.`.